packages/client/lib/AccessTokenClient/IssuanceInitiationAccessTokenClient.ts

```diff
@@ -13,38 +13,55 @@ import {
 import { ObjectUtils } from '@sphereon/ssi-types';
 import Debug from 'debug';
 
-import { MetadataClient } from './MetadataClient';
-import { convertJsonToURI, formPost } from './functions';
+import { IssuanceInitiationClient } from '../CredentialOffer';
+import { MetadataClient } from '../MetadataClient';
+import { convertJsonToURI, formPost } from '../functions';
+
+import { AccessTokenClient } from './AccessTokenClient';
 
 const debug = Debug('sphereon:openid4vci:token');
 
-export class AccessTokenClient {
-  public async acquireAccessTokenUsingIssuanceInitiation({
-    issuanceInitiation,
-    asOpts,
-    pin,
-    codeVerifier,
-    code,
-    redirectUri,
-    metadata,
-  }: AccessTokenRequestOpts): Promise<OpenIDResponse<AccessTokenResponse>> {
-    const { issuanceInitiationRequest } = issuanceInitiation;
+export class IssuanceInitiationAccessTokenClient implements AccessTokenClient {
+  getAccessTokenRequest = (
+    issuanceInitiationClient: IssuanceInitiationClient,
+    metadata: EndpointMetadata,
+    pin?: string,
+    clientId?: string,
+    codeVerifier?: string,
+    code?: string,
+    redirectUri?: string
+  ): IssuanceInitiationAccessTokenRequestOpts => {
+    return {
+      issuanceInitiation: issuanceInitiationClient.issuanceInitiationWithBaseUrl,
+      metadata: metadata,
+      pin,
+      codeVerifier,
+      code,
+      redirectUri,
+      asOpts: { clientId: clientId },
+    };
+  };
 
-    const isPinRequired = this.isPinRequiredValue(issuanceInitiationRequest);
-    const issuerOpts = { issuer: issuanceInitiationRequest.issuer };
+  public async acquireAccessToken(accessTokenRequestOpts: IssuanceInitiationAccessTokenRequestOpts): Promise<OpenIDResponse<AccessTokenResponse>> {
+    const accessTokenRequestOptions = accessTokenRequestOpts as IssuanceInitiationAccessTokenRequestOpts;
+    const issuanceInitiation = accessTokenRequestOptions.issuanceInitiation;
+    const req: IssuanceInitiationRequestPayload = issuanceInitiation.issuanceInitiationRequest;
+
+    const isPinRequired = this.isPinRequiredValue(req);
+    const issuerOpts = { issuer: req.issuer };
 
     return await this.acquireAccessTokenUsingRequest({
       accessTokenRequest: await this.createAccessTokenRequest({
-        issuanceInitiation,
-        asOpts,
-        codeVerifier,
-        code,
-        redirectUri,
-        pin,
+        issuanceInitiation: issuanceInitiation,
```

Comment:
Please just stick to the assignment at hand. The previous was more legable  as it used destructuring, so you do not get cluttered types that do not add much, as you are basically taking the inputs and using them again as arguments for a next function

Reply:
fixed